aws/data_source_aws_codecommit_repository_test.go

```diff
@@ -0,0 +1,46 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccAWSCodeCommitRepositoryDataSource_basic(t *testing.T) {
+	rName := fmt.Sprintf("tf-acctest-%d", acctest.RandInt())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAwsCodeCommitRepositoryDataSourceConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("data.aws_codecommit_repository.default", "repository_name",
+						"aws_codecommit_repository.default", "repository_name"),
+					resource.TestMatchResourceAttr("data.aws_codecommit_repository.default", "arn",
```

Comment:
Minor nitpick: I will update the rest of the checks to `resource.TestCheckResourceAttrPair()` post merge.